### ISSUE TITLE
humanize doesn't respect Infector acronyms

### ISSUE THREAD

**Issue by mintuhouse** [rails/rails#17879]
I added the API acronym using

```
ActiveSupport::Inflector.inflections do |inflect|
  inflect.acronym('API')
  inflect.acronym('APIs')
end
```

Console Output

```
"api".humanize 
=> "API"   # As expected

"API".humanize 
=> "Api"    # I was expecting "API" here too

```

Changing [this line](https://github.com/rails/rails/blob/36019329b015e570a58d9f34027c6e73edc36b77/activesupport/lib/active_support/inflector/methods.rb#L118) to following might do the trick.

```
 "#{inflections.acronyms[match.downcase] || match.downcase}"
```

**Comment by matthewd** [comment]
`"API"` doesn't sound like it really meets the definition of `lower_case_and_underscored_word`.

If you're passing things that look like constant names, you're generally unlikely to have a good time:

``` ruby
'foo_bar'.humanize # => 'Foo bar'
'FooBar'.humanize  # => 'Foobar'
```

Adding the `downcase` here doesn't appear likely to hurt anyone, but it does seem like a bit of a futile gesture towards supporting a usage that just can't work properly.

**Comment by recursive-madman** [comment]
An alternative would be to use `underscore` instead of `downcase`:

```ruby
'foo_bar'.underscore.humanize #=> "Foo bar"
'FooBar'.underscore.humanize #=> "Foo bar"
'api controller'.underscore.humanize #=> "API controller"
'API controller'.underscore.humanize #=> "API controller"
```

**Comment by mintuhouse** [comment]
@matthewd Sorry, I overlooked the function definition. 
Adding underscore/downcase would solve this issue. Thanks.

**Comment by kamipo** [comment]
Fixed by 0ddde0a8fca6a0ca3158e3329713959acd65605d.
